ccloader/js/mod.js

```diff
@@ -149,6 +183,79 @@ export class Mod {
 		this.runtimeAssets[original] = newPath;
 	}
 
+	/**
+	 * Returns the full path to a resource
+	 * @param {string} relativePath to resource with mod folder as base
+	 * @returns {string} 
+	 */
+	absolutePath(relativePath) {
+		const basePath = this._getBaseName(this.file); 
+		if (!relativePath) {
+			return this._normalizePath(basePath);
+		}
+
+		if (navigator.platform === 'Win32') {
+			// convert all \ to / and remove all duplicate //
+			relativePath = relativePath.replace(/\\/g, path.sep).replace(/\/\//g, path.sep);
+		}
+
+		// should protect against escaping mod directory
+		relativePath = path.normalize('/' + relativePath);
+		return this._normalizePath(basePath + relativePath);
+	}
+
+
+	/**
+	 * 
+	 * @param {string} relativePath to resource with mod folder as base
+	 * @returns {boolean} 
+	 */
+	hasResource(relativePath) {
+		if (!this.ready) {
+			return false;
+		}
+
+		for (const asset of this.assets) {
+			if (asset.endsWith(relativePath)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * 
+	 * @param {string} relativePath to resource with mod folder as base
+	 * @returns {any} 
+	 */
+	async getResource(relativePath) {
+		if (!this.ready) {
+			throw Error(`Cannot get resource from Mod "${this.manifest.name}". Mod is not ready.`);
```

Comment:
Ideally, if the mod is enabled but not loaded yet, this would wait until it is loaded and then execute the request.

Reply:
The mod may never be ready.

Reply:
I am not going to stall until it is ready.